gatsby/create-pages.js

```diff
@@ -57,10 +62,40 @@ function createCaseStudyPages(graphql, createPage, config) {
   });
 }
 
+function createBlogPostPages(graphql, createPage, config) {
+  return graphql(queryBlogPosts).then(result => {
+    if (result.errors) return Promise.reject(new Error(result.errors));
+    const getPath = replaceSlug(config.templates.blogPost.path);
+
+    const blogPosts = result.data.allContentfulPost.edges;
+
+    blogPosts.forEach(({ node: post }) => {
+      const pageConfig = {
+        path: getPath(post.slug),
+        component: path.resolve(
+          __dirname,
+          '../src/',
+          config.templates.blogPost.component,
+        ),
+        context: {
+          slug: post.slug,
+        },
+      };
+
+      createPage(pageConfig);
+    });
+
+    return Promise.resolve();
+  });
+}
+
 module.exports = ({ graphql, boundActionCreators }) => {
   const { createPage } = boundActionCreators;
 
   return getConfiguration(graphql).then(config =>
-    createCaseStudyPages(graphql, createPage, config),
+    Promise.all([
+      createCaseStudyPages(graphql, createPage, config),
```

Comment:
Let's not generate the blog pages in production, as they won't be a part of the MVP. We can do a sort of feature flip here.